examples/layer-shell.c

```diff
@@ -453,8 +454,11 @@ static void handle_global(void *data, struct wl_registry *registry,
 				&wl_seat_interface, 1);
 		wl_seat_add_listener(seat, &seat_listener, NULL);
 	} else if (strcmp(interface, zwlr_layer_shell_v1_interface.name) == 0) {
+		if (zwlr_layer_shell_v1_interface.version < (int)version) {
```

Comment:
Why the cast to `int`?

We can't auto-magically use `zwlr_layer_shell_v1_interface.version` here. If there is ever a v5, we don't want to automatically use v5 when re-compiling this example. We should explicitly cap the version with 4 instead.